### PR TITLE
Add skip_test env for custom build script

### PR DESCRIPTION
**Pull request by azhao155**
Tested

skip-tests=true
```
zyanshu@zyanshu-ubuntu:~/skaffold/examples/custom$ ../../out/skaffold build --skip-tests=true        
Generating tags...                                                                                                                                                                                                
 - github.com/googlecontainertools/skaffold/examples/custom -> github.com/googlecontainertools/skaffold/examples/custom:latest                                                                                    
Checking cache...                                                                                                                                                                                                 
 - github.com/googlecontainertools/skaffold/examples/custom: Not found. Building                         
Starting build...                                                                                                                                                                                                 
Found [minikube] context, using local docker daemon.                                                                                                                                                              
Building [github.com/googlecontainertools/skaffold/examples/custom]...                                   
SKIP_TEST: true                               
```

skip-tests=false
```
zyanshu@zyanshu-ubuntu:~/skaffold/examples/custom$ ../../out/skaffold build --skip-tests=false                                                                                                                    
Generating tags...                                                                                       
 - github.com/googlecontainertools/skaffold/examples/custom -> github.com/googlecontainertools/skaffold/examples/custom:latest                                                                                    
Checking cache...                                                                                        
 - github.com/googlecontainertools/skaffold/examples/custom: Not found. Building                                                                                                                                  
Starting build...                                                                                                                                                                                                 
Found [minikube] context, using local docker daemon.                                                                                                                                                              
Building [github.com/googlecontainertools/skaffold/examples/custom]...                                   
SKIP_TEST: false     
```

### CODE REVIEW

**Comment by azhao155** [comment]
resolving https://github.com/GoogleContainerTools/skaffold/issues/5909

File: examples/custom/build.sh [hunk 1]
```diff
@@ -1,6 +1,8 @@
 #!/usr/bin/env bash
 set -e
 
+echo "SKIP_TEST:" ${SKIP_TEST}
```
Comment (gsquared94) [inline examples/custom/build.sh:4]: can we revert this change to the example project? It does demo this PR working, but not relevant to this example.

File: pkg/skaffold/build/cluster/cluster.go [hunk 2]
```diff
@@ -82,7 +82,7 @@ func (b *Builder) runBuildForArtifact(ctx context.Context, out io.Writer, a *lat
 		return b.buildWithKaniko(ctx, out, a.Workspace, a.ImageName, a.KanikoArtifact, tag, requiredImages)
 
 	case a.CustomArtifact != nil:
-		return custom.NewArtifactBuilder(nil, b.cfg, true, append(b.retrieveExtraEnv(), util.EnvPtrMapToSlice(requiredImages, "=")...)).Build(ctx, out, a, tag)
+		return custom.NewArtifactBuilder(nil, b.cfg, true, false, append(b.retrieveExtraEnv(), util.EnvPtrMapToSlice(requiredImages, "=")...)).Build(ctx, out, a, tag)
```
Comment (gsquared94) [inline pkg/skaffold/build/cluster/cluster.go:85]: instead of `false` always read from `b.cfg.SkipTests()`. You'll have to add the method to the `Config` interface but it should already be implemented by `b.cfg`.

**Review by gsquared94 — APPROVED** [review]
lgtm!
